src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -174,6 +179,12 @@ public override void VisitParameterDeclarationSyntax(ParameterDeclarationSyntax
                     {
                         assignedType = UnionType.Create(allowedItemTypes);
                     }
+                    else if (object.ReferenceEquals(declaredType, LanguageConstants.String))
```

Comment:
Do you need the condition on 182? On line 173, you check if `assignedType` is the same object as the String type. At that time, `assignedType` is `declaredType` and `declaredType` hasn't changed since.

Reply:
Good point!